internal/reader/rmn_home.go

```diff
@@ -0,0 +1,329 @@
+)
+
+type RMNHome interface {
+	// GetRMNNodesInfo gets the RMNHomeNodeInfo for the given configDigest
+	GetRMNNodesInfo(configDigest cciptypes.Bytes32) ([]rmntypes.RMNHomeNodeInfo, error)
+	// IsRMNHomeConfigDigestSet checks if the configDigest is set in the RMNHome contract
+	IsRMNHomeConfigDigestSet(configDigest cciptypes.Bytes32) bool
+	// GetMinObservers gets the minimum number of observers required for each chain in the given configDigest
+	GetMinObservers(configDigest cciptypes.Bytes32) (map[cciptypes.ChainSelector]uint64, error)
+	// GetOffChainConfig gets the offchain config for the given configDigest
+	GetOffChainConfig(configDigest cciptypes.Bytes32) (cciptypes.Bytes, error)
+	services.Service
+}
+
+type rmnHomeState struct {
+	primaryConfigDigest   cciptypes.Bytes32
+	secondaryConfigDigest cciptypes.Bytes32
+	rmnHomeConfig         map[cciptypes.Bytes32]rmntypes.RMNHomeConfig
+}
+
+// RmnHomePoller polls the RMNHome contract for the latest RMNHomeConfigs
+// It is running in the background with a polling interval of pollingDuration
+type RmnHomePoller struct {
+	wg                   sync.WaitGroup
+	stopCh               services.StopChan
+	sync                 services.StateMachine
+	mutex                *sync.RWMutex
+	contractReader       contractreader.ContractReaderFacade
+	rmnHomeBoundContract types.BoundContract
+	lggr                 logger.Logger
+	rmnHomeState         rmnHomeState
+	failedPolls          uint
+	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
+}
+
+func NewRMNHomePoller(
+	contractReader contractreader.ContractReaderFacade,
+	rmnHomeBoundContract types.BoundContract,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) RMNHome {
+	return &RmnHomePoller{
+		stopCh:               make(chan struct{}),
+		contractReader:       contractReader,
+		rmnHomeBoundContract: rmnHomeBoundContract,
+		rmnHomeState:         rmnHomeState{},
+		mutex:                &sync.RWMutex{},
+		failedPolls:          0,
+		lggr:                 lggr,
+		pollingDuration:      pollingInterval,
+	}
+}
+
+func (r *RmnHomePoller) Start(ctx context.Context) error {
+	r.lggr.Infow("Start Polling RMNHome")
+	return r.sync.StartOnce(r.Name(), func() error {
+		r.wg.Add(1)
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *RmnHomePoller) poll() {
+	defer r.wg.Done()
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	// Initial fetch once poll is called before any ticks
+	if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+		// Just log, don't return error as we want to keep polling
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+	}
+
+	ticker := time.NewTicker(r.pollingDuration)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			r.mutex.Lock()
+			r.failedPolls = 0
+			r.mutex.Unlock()
+			return
+		case <-ticker.C:
+			if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+				r.mutex.Lock()
+				r.failedPolls++
```

Comment:
Should this be reset if `err == nil`?

Reply:
We are reporting the failed report in a raw, so you are correct. Good catch.